ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoPlayer.swift

```diff
@@ -0,0 +1,292 @@
+
+    playStartedEvent?()
+  }
+
+  func maybeCancelPlay() {
+    if !isAutoplayFinished && isLoadFinished {
+      autoplayFinished()
+      player?.pause()
+    }
+
+    if isPlayStarted {
+      player?.pause()
+      if let timeObserver = timeObserver {
+        player?.removeTimeObserver(timeObserver)
+      }
+      timeObserver = nil
+
+      playCancelledEvent?()
+    }
+    isPlayStarted = false
+  }
+
+  func togglePlay() -> Bool {
+    let isVideoInProgress = isVideoInProgress()
+
+    if isVideoInProgress {
+      player?.pause()
+    } else {
+      player?.play()
+    }
+
+    return !isVideoInProgress
+  }
+
+  func maybeStartAutoplay() {
+    // Start autoplay only once.
+    if isAutoplayRequestedOnce {
+      return
+    }
+    isAutoplayRequestedOnce = true
+
+    // Start autoplay after video is loaded
+    if !isLoadFinished {
+      return
+    }
+
+    startAutoplay()
+  }
+
+  private func loadVideoTrackParams(_ backgroundVideoPath: URL) {
+    Task {
+      var frameRate: Float?
+      let asset: AVURLAsset = AVURLAsset(url: backgroundVideoPath)
+      let isPlayable = try? await asset.load(.isPlayable)
+      if let isPlayable = isPlayable,
+        isPlayable,
+        let videoTrack = try? await asset.loadTracks(withMediaType: .video).first
+      {
+        frameRate = try? await videoTrack.load(.nominalFrameRate)
+      }
+
+      finishPlayerSetup(isPlayable: isPlayable, frameRate: frameRate)
+    }
+  }
+
+  private func finishPlayerSetup(isPlayable: Bool?, frameRate: Float?) {
+    DispatchQueue.main.async { [weak self] in
+      guard let self = self else { return }
+
+      guard let isPlayable = isPlayable else {
+        self.videoLoaded(succeeded: false)
+        return
+      }
+      self.frameRate = frameRate
+      self.videoLoaded(succeeded: isPlayable)
+    }
+  }
+
+  private func startAutoplay() {
+    if isAutoplayFinished {
+      autoplayFinished()
+      return
+    }
+
+    guard let duration = player?.currentItem?.duration else {
+      autoplayFinished()
+      return
+    }
+
+    var autoplayLengthSeconds = self.kMaxAutoplayDuration
+    if let frameRate = frameRate,
+      let stopFrame = stopFrame
+    {
+      let stopFrameTime = Double(stopFrame) / Double(frameRate)
+      if stopFrameTime < autoplayLengthSeconds {
+        autoplayLengthSeconds = stopFrameTime
+      }
+    }
+
+    // If autoplay length is less then video duration then set forward playback end
+    // time to stop the video at the specified time.
+    if autoplayLengthSeconds < CMTimeGetSeconds(duration) {
+      player?.currentItem?.forwardPlaybackEndTime = CMTime(
+        seconds: autoplayLengthSeconds,
+        preferredTimescale: CMTimeScale(NSEC_PER_SEC)
+      )
+    }
+
+    // Detect if video was interrupted during autoplay and fire autoplayFinished
+    // event if needed.
+    playerObserver = player?.observe(\.timeControlStatus, options: [.new, .old]) {
+      [weak self] (player, change) in
```

Comment:
I dont believe you need `.new` or `.old` at all since you're not reading from the `change` dictionary, you also can omit the change variable entirely (replace with `_`) since you never ref it